src/CurlUtil.cc

```diff
@@ -199,21 +200,33 @@ void CurlWorker::Run() {
 	CURLMcode mres = CURLM_OK;
 
 	std::vector<struct curl_waitfd> waitfds;
-	waitfds.resize(1);
+	waitfds.resize(2);
```

Comment:
I'll add a comment.

Basically, these two file descriptors become readable when there's additional work to do.  This triggers the sleeping libcurl to wake up and, eventually, call `m_queue->TryConsume` / `m_unpause_queue->TryConsume` (which will then read from the file descriptors.

These are needed because we have both I/O calls and shared memory condition variables but the various `select()`-like APIs won't wake up with the condition variable is ready.